roles/docker-compose/tasks/install_docker.yml

```diff
@@ -0,0 +1,57 @@
+---
+
+- name: Install docker dependencies
+  apt:
+    name: '{{ packages }}'
+    update_cache: yes
+    cache_valid_time: 3600
+  vars:
+    packages:
+    - apt-transport-https
+    - ca-certificates
+    - curl
+    - software-properties-common
+
+- name: Add Docker’s official GPG key
+  apt_key:
+    url: https://download.docker.com/linux/ubuntu/gpg
+    state: present
+
+- name: Add Docker's official APT repository
+  apt_repository:
+    repo: deb [arch=amd64] https://download.docker.com/linux/ubuntu {{ansible_distribution_release}} stable
+    state: present
+    update_cache: yes
+
+- name:
+  apt:
+    name: '{{ item }}'
+    update_cache: yes
+    cache_valid_time: 3600
+  with_items:
```

Comment:
This is probably unnecessary for one-item lists + should be moved to name anyway